mep-0008.md

```diff
@@ -0,0 +1,484 @@
+ - Users have explicit control of `app.function`.
+
+**Disadvantages:**
+ - Inexperienced users may get stuck in the mode and not understand why the cell doesn't run.
+ - Possible to save a function with a reference to something else in the notebook, which will cause a `NameError` when actually run.  
+   This could be mitigated by a warning when saving but may still be confusing.
+ - Additional metadata on the cell level is required to distinguish between the two modes.
+
+**UX Suggestion:**
+
+![Shows option to move to function mode](https://github.com/user-attachments/assets/8618816c-2a3e-4e3b-b3dd-776f3c354bc2)
+
+Just another "mode":
+
+![Shows function mode](https://github.com/user-attachments/assets/3d0ebef5-e2e2-40a5-99c1-fccc220627f3)
+
+---
+
+**Suggestion:**  
+ - Let's stick to **unmanaged** for now, as it's consistent with how Markdown and SQL cells work.
+
+---
+
+## Serialization
+
+As noted above, the import block serialization is not yet settled.  
+Here are a couple of suggestions:
+
+### Comment Separator
+
+```python
+# These imports are auto-generated by marimo.
+# Try modifying the source cell definitions,
+# as opposed to the following block.
+import pytest
+
+import marimo
+
+__generated_with = "0.0.0"
+# ...
+```
+
+[Working implementation and relevant test](https://github.com/marimo-team/marimo/pull/3779/files/db9d2578625f7b9d39d3dafa27b85d4329a81673#diff-339016dabb337751a0636e1c054e0fb97a11a6dd09185845dde365f47652b3d3R3-R8)
+
+**Pros:**
+ - Looks nice.
+ - No linter issues.
+
+**Cons:**
+ - Import order does not ensure `marimo` is first on format.
+ - Brittle detection.
+
+### `if` Separator
+
+```python
+import marimo
+
+if marimo.import_guard():
+    # This section is auto-generated by marimo.
+    import io
+    import textwrap
+    import marimo as mo
+    import typing
+    from pathlib import Path
+
+__generated_with = "0.0.0"
+```
+
+[Working implementation and relevant test](https://github.com/marimo-team/marimo/pull/3779/commits/7db678f9754dde74d8de3552dec6fd4f531bc1dc#diff-f0ce04493f0124e1c100939808ef5291bd6df9b32724091bbc9e360f29a6fb22R7-R15)
+
+**Pros:**
+ - Conditionally allows imports.
+ - Easy to detect.
+
+**Cons:**
+ - All or nothing.
+ - Some linters may complain that the imports may not be defined.
+
+### `with` Separator
+
+```python
+import marimo
+
+with marimo.import_guard():
+    # These imports are auto-generated by marimo.
+    # Try modifying the source cell definitions
+    # as opposed to the following block.
+    import pytest
+
+__generated_with = "0.0.0"
+```
+
+[Working implementation and relevant test](https://github.com/marimo-team/marimo/pull/3779/commits/36aa98e56daf687d4d89115e409dc744cdd9a0b3#diff-f0ce04493f0124e1c100939808ef5291bd6df9b32724091bbc9e360f29a6fb22R7-R17)
+
+**Pros:**
+ - Granularly allows imports.
+ - Easy to detect.
+
+**Cons:**
+ - `with` block skips are a bit of a hack and may be confusing.
+ - Some linters might complain about shadowed/duplicate imports.
+
```

Comment:
All things considered, do you think we should use `if` or `with`?

Reply:
I'm not convinced on the "selective" import, but I think by virtue of `with` having no linter issues, it wins out.

Maybe a little strange, but it's a unique use case. I think we get to set the precedent here